### ISSUE TITLE
Register goes above FFh

### ISSUE THREAD

**Issue by ajhewitt** [rokkerruslan/i8080#3]
The `H` register will count to `FFh` and then `100h` and continue counting without setting the `zero` flag and incrementing the `A` register.

```asm
            LXI     H, F0F0h
    LOOP:
            MOV     A, M
            INR     H
            JNZ     LOOP
            INR     A
            JMP     LOOP
```

**Comment by ajhewitt** [comment]
Closer examination of the code reveals the issue here: The emulator registers are not bound to their data types. The "8-bit" registers can be updated with values > `100h`, and the "16-bit" registers > `10000h`. The registers can also contain negative numbers (not 2's compliment of their data types). This fundamentally breaks the emulation.

**Comment by rokkerruslan** [comment]
You are right ;) Fixed here https://github.com/rokkerruslan/i8080/commit/c872a5502ae3211488d99cd445c28f06a8ba3930
